### ISSUE TITLE
Hello, how do you handle the data of Cifar10-dvs so that it has only one frame? thank you

### ISSUE THREAD

**Issue by hushengwang1999** [Dengyu-Wu/SNN-Regularisation-Cutoff#1]

**Comment by Dengyu-Wu** [comment]
Hi,

The experiment on Cifar10-DVS can be done as follows,

1. Compute the average spiking rate of training samples for ANN training, for example, the total events or spikes of each pixel are normalised by the maximum number of spikes. 
2. Convert ANN into SNN. 
3. Use SNN to directly process input spikes. For example,  forward the number of spikes over a specific time (from 0 to $t$) in SNN and examine the accuracy.

Hope the explanation above helps.

**Comment by hushengwang1999** [comment]
Thank you, but there is still a problem. When I used the ANN network framework you provided to train DVS-CIFAR10, the parameter settings were almost the same (because I used the Pytorch framework, which cannot be completely consistent), but the results differ greatly from what you provided, at least about 10 points. What is the reason for this? Thank you for your explanation.

**Comment by hushengwang1999** [comment]
Thank you for your continued response. The preprocessing method for my dataset is based on the Spikejelly framework and your paper. The cumulative number of pulses per data frame is calculated as input. The training method is similar to yours. The network structure and training method are shown in the following figure, and the random seed is also set to 1000. Thank you for your suggestion.
![图片](https://user-images.githubusercontent.com/51084088/228101074-9b1c741b-f343-4f4e-8bc4-f1f0c984ecbf.png)
![图片](https://user-images.githubusercontent.com/51084088/228101168-78a3fa55-4c78-4ae7-b020-a3f613112086.png)
![图片](https://user-images.githubusercontent.com/51084088/228101211-137bf348-9ad7-4deb-94a6-9eb8a972730f.png)
![图片](https://user-images.githubusercontent.com/51084088/228101830-0332bcdf-2bcc-47f4-ae6f-75353a913f75.png)
![图片](https://user-images.githubusercontent.com/51084088/228101860-16988447-56fe-4b7c-af9f-99f7d6a89d63.png)
![图片](https://user-images.githubusercontent.com/51084088/228101893-fc18b679-9956-46e1-aa55-51e558f0f1bb.png)
![图片](https://user-images.githubusercontent.com/51084088/228101976-552d47e5-6984-4e31-b116-b3837d0bfd83.png)

**Comment by Dengyu-Wu** [comment]
Can you check the size of x, after x.permute()?

**Comment by hushengwang1999** [comment]
Here, I select the number of data frames T=2, batch b=64, data polarity p=2, data frame height h=128, and width w=128. Therefore, the format of x is [64,2,2,128,128], and after permute(), it is converted to [2,64,2,128128], which is consistent with the code I gave myself.

**Comment by Dengyu-Wu** [comment]
Is the input normalised?

**Comment by hushengwang1999** [comment]
When the input is normalised, the result also drops.

**Comment by Dengyu-Wu** [comment]
There are several settings not same as our work, 

- Slightly different setting on network, e.g., kernel size, additional Dropout layer after FC layer, additional FC layer for voting, Batch Normalisation layer is missing after first FC layer. However, I do not think these differences would affect the training much, but small Dropout rate is always recommended. 
- The final training objective is different. Cifar10-DVS can use multiple frames as input, while our work uses single frame. For multiple frames, I would recommend to add Softmax before the summation of the output. For example, code in the forward function is updated as follows
```
for t in range(1,x.shape[0]):
  output_spikes += nn.Softmax(dim=1)(self.vote(self.fc(self.conv(x[t]))))
```
- Preprocessing is different, e.g., the events should be split by 'time' not 'number'. 

You can first update your code accordingly and see if that helps. It will take a while to ensure everything is consistent with our work as you are using Pytorch.

**Comment by Dengyu-Wu** [comment]
The processed data might be a bit large to share. 

I have updated the repository to include the preprocessing [code](https://github.com/Dengyu-Wu/SNN-Regularisation-Cutoff/blob/main/preprocessing/cifar10dvs.py). You can refer to the code to convert the event-based samples into frame-based samples on your machine.

**Comment by hushengwang1999** [comment]
thank you!

**Comment by hushengwang1999** [comment]
Hello, where is this file
![图片](https://user-images.githubusercontent.com/51084088/228830615-f2f3c5c1-ef5d-4754-a1bf-4d415791a54d.png)
? Thank you

**Comment by Dengyu-Wu** [comment]
I believe that I have addressed your concern with the response. I am now closing this issue. If you have any further questions or concerns related to this issue, please feel free to open a new issue
